src/mapblock.h

```diff
@@ -669,41 +669,36 @@ typedef std::vector<MapBlock*> MapBlockVect;
 
 inline bool objectpos_over_limit(v3f p)
 {
-	// MAP_BLOCKSIZE must be subtracted to avoid an object being spawned just
-	// within the map generation limit but in a block and sector that extend
-	// beyond the map generation limit.
-	// This avoids crashes caused by sector over limit in createSector().
-	const float object_limit = (MYMIN(MAX_MAP_GENERATION_LIMIT,
-		g_settings->getU16("map_generation_limit")) - MAP_BLOCKSIZE) * BS;
-	return (p.X < -object_limit
-		|| p.X > object_limit
-		|| p.Y < -object_limit
-		|| p.Y > object_limit
-		|| p.Z < -object_limit
-		|| p.Z > object_limit);
+	const float max_limit_bs = MAX_MAP_GENERATION_LIMIT * BS;
```

Comment:
l assume, if you make this static, it severely harms performance here.
A static variable is not saved on the stack, thus it explicitly needs to be loaded from memory. When the function is called shortly again, it can be loaded from the processor cache.
If this is not static, the compiler can optimize it to store the value in an immediate, register or whatsoever.
The values of MAX_MAP_GENERATION_LIMIT and BS are known at compile time, thus they are multiplied when compiling and multiplying doesn't happen at runtime.

The functions shouldn't explicitly be inlined, gcc automatically detects whether it improves performance by reducing jumps or bloats machine code, which exacerbates caching.